Updates/0.3.3.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\QueuedTracking;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\QueuedTracking\Queue\Manager;
+use Piwik\Updater;
+use Piwik\Updates;
+
+class Updates_0_3_3 extends Updates
+{
+    /**
+     * @var Manager
+     */
+    protected $queueManager;
+
+    public function doUpdate(Updater $updater)
+    {
+        $settings = $this->getSettings();
+        $numQueueWorkers = $settings->numQueueWorkers->getValue();
+
+        if ($numQueueWorkers <= 1) {
+            return; // it is pointless to rewrite one queue to the same queue
+        }
+
+        $this->rewriteQueues($numQueueWorkers);
```

Comment:
Looks like creating temporary queues is a good idea. This is how I see it.

We have QueueNameCreator class that:
* creates queue name based on distribution algorithm, number of queues and queue index, eg. `queue_crc32_4_0`,
* same as above, but with some prefix, eg. `tmp_queue_crc32_4_0`,
* creates legacy name of a queue based on queue index, eg. `trackingQueueV1`.

This way we have queue name generating that supports both distribution algorithm change and queues number change.

We have 2 major cases to consider:

1. **Distribution algorithm changes**. In this case distributing to new queues (`queue_crc32_4_0`, `queue_crc32_4_1`...) starts immediately after plugin's code is updated. Update class creates temporary queues (`tmp_queue_crc32_4_0`, `tmp_queue_crc32_4_1`...) and rewrites legacy queues (`trackingQueueV1`, `trackingQueueV1_1`...) into temporary ones. After that each working queue (`queue_crc32_4_0`...) is unioned to corresponding tmp queue (with `\Redis::sUnionStore()`), right after that working queue is deleted and tmp queue is renamed to working queue (with `\Redis::rename()`). After that legacy queues are removed.

This way:
* order of requests is preserved,
* only small numer of requests may be dropped between deleting old working queue and putting temporary queue in pace of it.

2. **Number of queues changes**. Looks like this case is analogical to previous one, regardless whether we decrease or increase number of queues. In both cases immediately after change of number of queues new set of working queues is created. Meanwhile legacy queues are redistributed into temporary ones and in the end temporary ones replace working ones. After that legacy queues are removed.

So looks like such algorithm looks universal for both major cases and could be implemented as a part of this plugin. It also could solve current problem that when number of queues is increased, some later requests in the visits will be processed before earlier ones. It also should require less memory, at least on PHP's side.